### PR TITLE
azurerm_kusto_eventhub_data_connection: Support for `compression`

### LINKED ISSUE #9363: Support compression for azurerm_kusto_eventhub_data_connection
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

[azurerm_kusto_eventhub_data_connection](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kusto_eventhub_data_connection) does not have support for `compression`. See [az kusto data-connection event-hub create](https://docs.microsoft.com/en-us/cli/azure/ext/kusto/kusto/data-connection/event-hub?view=azure-cli-latest#ext_kusto_az_kusto_data_connection_event_hub_create) `--compression`
```
az kusto data-connection event-hub create --cluster-name
                                          --data-connection-name
                                          --database-name
                                          --resource-group
                                          [--compression {GZip, None}]
                                          [--consumer-group]
                                          [--data-format {APACHEAVRO, AVRO, CSV, JSON, MULTIJSON, ORC, PARQUET, PSV, RAW, SCSV, SINGLEJSON, SOHSV, TSV, TSVE, TXT, W3CLOGFILE}]
                                          [--event-hub-resource-id]
                                          [--event-system-properties]
                                          [--location]
                                          [--mapping-rule-name]
                                          [--no-wait]
                                          [--table-name]
```
Our use case, Prometheus samples, has a compression ratio 10:1, so giving it up is not an option.

### New or Affected Resource(s)

* azurerm_kusto_eventhub_data_connection

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

### PR DESCRIPTION
**Pull request by favoretti**
azurerm_kusto_eventhub_data_connection: Support for `compression` property

Fixes #9363

### CODE REVIEW

File: azurerm/internal/services/kusto/kusto_eventhub_data_connection_resource.go [hunk 1]
```diff
@@ -55,6 +55,16 @@ func resourceArmKustoEventHubDataConnection() *schema.Resource {
 				ValidateFunc: validateAzureRMKustoClusterName,
 			},
 
+			"compression": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
```
Comment (tombuildsstuff) [inline azurerm/internal/services/kusto/kusto_eventhub_data_connection_resource.go:61]: is there a default value for this? I presume `None`?

**Review by tombuildsstuff — APPROVED** [review]
otherwise LGTM 👍

File: azurerm/internal/services/kusto/kusto_eventhub_data_connection_resource.go (hunk 1 above)
Comment (favoretti) [reply azurerm/internal/services/kusto/kusto_eventhub_data_connection_resource.go:61]: Seems like it. Set default to None and updated docs.

**Comment by tombuildsstuff** [comment]
Test Passes:

Screenshot 2020-12-07 at 18 03 01

**Pull request merged**